File: mobile/minibilge_mobile/lib/features/education/models/level.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'level.freezed.dart';
part 'level.g.dart';

@freezed
class Level with _$Level {
  const factory Level({
    @JsonKey(name: 'Id') required String id,
    @JsonKey(name: 'TopicId') required String topicId,
    @JsonKey(name: 'Name') required String name,
    @JsonKey(name: 'Description') String? description,
    @JsonKey(name: 'Difficulty') required int difficulty,
    @JsonKey(name: 'DisplayOrder') required int displayOrder,
    @JsonKey(name: 'MinCorrectToPass') required int minCorrectToPass,
    @JsonKey(name: 'IsActive') required bool isActive,
  }) = _Level;

  factory Level.fromJson(Map<String, dynamic> json) =>
      _$LevelFromJson(json);
}

File: backend/MiniBilge.Domain/Entities/Level.cs
using MiniBilge.Domain.Entities.Base;

namespace MiniBilge.Domain.Entities;

public class Level : BaseEntity
{
    public Guid TopicId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int Difficulty { get; set; } // 1-10 arası zorluk seviyesi
    public int DisplayOrder { get; set; }
    public int MinCorrectToPass { get; set; } = 7; // Geçmek için minimum doğru sayısı (10 sorudan)
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public Topic Topic { get; set; } = null!;
    public ICollection<Question> Questions { get; set; } = new List<Question>();
}
